Add Contributing gude (#21)
jenkinsci/discard-old-build-plugin: CONTRIBUTING.md

# Contributing to the built on column plugin

Plugin source code is hosted on [GitHub](https://github.com/jenkinsci/discard-old-build-plugin).
New feature proposals and bug fix proposals should be submitted as
[GitHub pull requests](https://help.github.com/articles/creating-a-pull-request).

> Your pull request will be evaluated by the [Jenkins job](https://ci.jenkins.io/job/Plugins/job/built-on-column-plugin/).

Reviewer: Use correct URL for Jenkins job.
```suggestion
Your pull request will be evaluated by the [Jenkins job](https://ci.jenkins.io/job/Plugins/job/discard-old-build-plugin/).
```

  If a plugin maintainer wants to use a different Jenkins version, they run the command:

```bash
  mvn -Djenkins.version=2.375.1 hpi:run
```

* If a plugin maintainer wants to use a different HTTP port, (as in http://localhost:9090/jenkins) they run the command:

```bash
  mvn -Dport=9090 hpi:run
```

* If a plugin maintainer wants to allow other computers to access that Jenkins controller while it is running, they run the command:

```bash
  mvn -Dhost=0.0.0.0 hpi:run
```

## Code Coverage

[JaCoCo code coverage](https://www.jacoco.org/jacoco/) reporting is available as a maven target and can be displayed by the [Jenkins warnings next generation plugin](https://plugins.jenkins.io/warnings-ng/).
Please try to improve code coverage with tests when you submit.
* `mvn -P enable-jacoco clean install jacoco:report` to report code coverage with JaCoCo.

Please don't introduce new spotbugs output.
* `mvn spotbugs:check` to analyze project using [Spotbugs](https://spotbugs.github.io)
* `mvn spotbugs:gui` to review report using GUI

## Maintaining automated tests

Automated tests are run as part of the `verify` phase.

> Automated tests in the `continuous-integration` profile are run with multiple Java virtual machines, depending on the number of available processor cores.

Reviewer: Not for this plugin.  This requires a specific profile in the pom.xml file that is not included in this plugin.  That profile is included in some plugins but not in all plugins.
```suggestion
```

Run automated tests with multiple Java virtual machines in a development with the command:

```
$ mvn clean -DforkCount=1C verify
```

## Report an Issue

Use the ["Report an issue" page](https://www.jenkins.io/participate/report-issue/redirect/#16001) to submit bug reports.

## Security Issues

Follow the [Jenkins project vulnerability reporting instructions](https://jenkins.io/security/reporting/) to report vulnerabilities.

> Footer
> © 2022 GitHub, Inc.
> Footer navigation
> Terms
> Privacy
> Security
> Status

Reviewer: Please remove this extra text.
```suggestion
```
